### PR TITLE
[cr] Find all references in the `RelationalView`

### PR DESCRIPTION
**Pull request by decentralion**
This pull request renames `findReferences` to `parseReferences`, and then implements `findReferences` as a module that finds every reference in a `RelationalView`. This will make it very easy to add reference edges to the GitHub graph.

See individual commits for details.

### CODE REVIEW

File: src/v3/plugins/github/findReferences.js [hunk 1]
```diff
@@ -1,58 +1,63 @@
...
-}
+export type GithubReference = {|
+  src: N.TextContentAddress,
+  dst: N.ReferentAddress,
+|};
 
-function findNumericReferences(body: string): string[] {
-  return findAllMatches(/(?:\W|^)(#\d+)(?:\W|$)/g, body).map((x) => x[1]);
-}
+type TextContentEntry =
```
Comment (wchargin) [inline src/v3/plugins/github/findReferences.js]: Should this be defined in the `relationalView.js` module?

File: src/v3/plugins/github/findReferences.test.js [hunk 2]
```diff
@@ -2,130 +2,13 @@
...
-      )
-    ).toEqual([
-      "#125",
-      "https://github.com/sourcecred/sourcecred/pull/125#pullrequestreview-113402856",
-      "@wchargin",
-    ]);
+import * as R from "./relationalView";
+
+describe("plugins/github/findReferences", () => {
+  it("matches snapshot on exmaple-github data", () => {
```
Comment (wchargin) [inline src/v3/plugins/github/findReferences.test.js]: nit: _example_ (sp.)

File: src/v3/plugins/github/findReferences.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  function* allGithubEntities(): Iterator<R.Entry> {
+    yield* view.repos();
+    yield* view.issues();
+    yield* view.pulls();
+    yield* view.reviews();
+    yield* view.comments();
+    yield* view.userlikes();
+  }
+
+  const refToAddress: Map<string, N.StructuredAddress> = new Map();
```
Comment (wchargin) [inline src/v3/plugins/github/findReferences.js]: I don’t understand why this is called `refToAddress` and not
`urlToAddress`. It doesn’t map from `GithubReference`s. What
am I missing?

Comment (wchargin) [inline src/v3/plugins/github/findReferences.js]: I don’t understand why this is called `refToAddress` and not
`urlToAddress`. It doesn’t map from `GithubReference`s. What
am I missing?

File: src/v3/plugins/github/findReferences.test.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// @flow
+
+import {findReferences} from "./findReferences.js";
+
+import * as R from "./relationalView";
+
+describe("plugins/github/findReferences", () => {
+  it("matches snapshot on exmaple-github data", () => {
+    const data = require("./demoData/example-github");
```
Comment (wchargin) [inline src/v3/plugins/github/findReferences.test.js]: Do you want a type assertion here, or do you find it superfluous?

File: src/v3/plugins/github/findReferences.js (hunk 3 above)
Comment (wchargin) [inline src/v3/plugins/github/findReferences.js]: I don’t understand why this is called `refToAddress` and not
`urlToAddress`. It doesn’t map from `GithubReference`s. What
am I missing?

File: src/v3/plugins/github/findReferences.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      case "USERLIKE":
+        refToAddress.set(`@${e.address.login}`, a);
+        break;
+      case "ISSUE":
+        refToAddress.set(`#${e.address.number}`, a);
+        break;
+      case "PULL":
+        refToAddress.set(`#${e.address.number}`, a);
+        break;
+      default:
```
Comment (wchargin) [inline src/v3/plugins/github/findReferences.js]: Consider adding here explicit cases for each other possible value of
`e.type`, and then using an `empty`-assertion in the `default`-case.
That way, when we add new GitHub node types, we’ll be forced to consider
this block, and either add a new reference type or explicitly decline to
do so.

File: src/v3/plugins/github/findReferences.js (hunk 3 above)
Comment (decentralion) [reply src/v3/plugins/github/findReferences.js]: Made a clearer docstring in the new version, but:
"# 34", "@ wchargin" and "https://github.com/ sourcecred/sourcecred/pull/416" are all considered refs.

File: src/v3/plugins/github/findReferences.js (hunk 5 above)
Comment (decentralion) [reply src/v3/plugins/github/findReferences.js]: Done, in the new version.

**Pull request closed without merging**